atc/db/team_test.go

```diff
@@ -2161,6 +2161,262 @@ var _ = Describe("Team", func() {
+
+					err = resourceScope.SaveVersions(nil, []atc.Version{
+						{"disabled": "version"},
+					})
+					Expect(err).ToNot(HaveOccurred())
+
+					rcv, found, err = resourceScope.FindVersion(atc.Version{"disabled": "version"})
+					Expect(err).ToNot(HaveOccurred())
+					Expect(found).To(BeTrue())
+
+					err = resource.DisableVersion(rcv.ID())
+					Expect(err).ToNot(HaveOccurred())
+
+					versions, _, found, err := resource.Versions(db.Page{Limit: 3}, nil)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(found).To(BeTrue())
+					Expect(versions).To(HaveLen(1))
+					Expect(versions[0].Version).To(Equal(atc.Version{"disabled": "version"}))
+					Expect(versions[0].Enabled).To(BeFalse())
+				})
+
+				It("should successfully update resource name", func() {
+
+					config.Resources[0].Name = "disabled-resource"
+					config.Resources[0].OldName = "some-resource"
+					config.Jobs[0].PlanSequence = []atc.Step{
+						{
+							Config: &atc.GetStep{
+								Name:     "some-input",
+								Resource: "disabled-resource",
+								Params: atc.Params{
+									"some-param": "some-value",
+								},
+								Passed:  []string{"job-1", "job-2"},
+								Trigger: true,
+							},
+						},
+					}
+
+					pipeline, _, err = team.SavePipeline(pipelineName, config, pipeline.ConfigVersion(), false)
+					Expect(err).ToNot(HaveOccurred())
+
+					updatedResource, _, _ := pipeline.Resource("disabled-resource")
+					Expect(updatedResource.ID()).To(Equal(resource.ID()))
+				})
+			})
+
+			Context("when new resource exists but the version is pinned", func() {
+				var pipeline db.Pipeline
+				var resource db.Resource
+				var pinnedVersion atc.Version
+				var err error
+
+				BeforeEach(func() {
+					pipeline, _, err = team.SavePipeline(pipelineName, config, 0, false)
+					Expect(err).ToNot(HaveOccurred())
+
+					resource, _, _ = pipeline.Resource("some-resource")
+
+					setupTx, err := dbConn.Begin()
+					Expect(err).ToNot(HaveOccurred())
+
+					brt := db.BaseResourceType{
+						Name: "some-type",
+					}
+
+					_, err = brt.FindOrCreate(setupTx, false)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(setupTx.Commit()).To(Succeed())
+
+					resourceScope, err := resource.SetResourceConfig(atc.Source{"some": "other-repository"}, atc.VersionedResourceTypes{})
+					Expect(err).ToNot(HaveOccurred())
+
+					err = resourceScope.SaveVersions(nil, []atc.Version{
+						atc.Version{"version": "v1"},
+						atc.Version{"version": "v2"},
+						atc.Version{"version": "v3"},
+					})
+					Expect(err).ToNot(HaveOccurred())
+
+					pinnedVersion = atc.Version{"version": "v1"}
+					resConf, found, err := resourceScope.FindVersion(pinnedVersion)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(found).To(BeTrue())
+					resID := resConf.ID()
+
+					found, err = resource.PinVersion(resID)
+					Expect(found).To(BeTrue())
+					Expect(err).ToNot(HaveOccurred())
+				})
+
+				It("should successfully update resource name", func() {
```

Comment:
This tests the right thing (i.e. that the `APIPinnedVersion()` remains unchanged), but maybe you should change the test description? e.g.

```suggestion
				It("should not change the pinned version", func() {
```